qiita_ware/test/test_processing_pipeline.py

```diff
@@ -271,6 +325,54 @@ def test_get_preprocess_fastq_cmd(self):
         self.assertEqual(obs_cmd_1, exp_cmd_1)
         self.assertEqual(obs_cmd_2, exp_cmd_2)
 
+    def test_get_preprocess_fastq_cmd_per_sample_FASTQ(self):
+        metadata_dict = {
+            'SKB8.640193': {'run_prefix': "sample1", 'primer': 'A',
+                            'barcode': 'A', 'center_name': 'ANL',
+                            'platform': 'ILLUMINA',
+                            'library_construction_protocol': 'A',
+                            'experiment_design_description': 'A'},
+            'SKD8.640184': {'run_prefix': "sample2", 'primer': 'A',
+                            'barcode': 'A', 'center_name': 'ANL',
+                            'platform': 'ILLUMINA',
+                            'library_construction_protocol': 'A',
+                            'experiment_design_description': 'A'}}
+        md_template = pd.DataFrame.from_dict(metadata_dict, orient='index')
+        prep_template = PrepTemplate.create(md_template, Study(1), '16S')
+
+        tmp_dir = mkdtemp()
+        self.dirs_to_remove.append(tmp_dir)
+        self.path_builder = partial(join, tmp_dir)
+        fp1 = self.path_builder('sample1.fastq')
+        with open(fp1, 'w') as f:
+            f.write('\n')
+        self.files_to_remove.append(fp1)
+        fp2 = self.path_builder('sample2.fastq.gz')
+        with open(fp2, 'w') as f:
+            f.write('\n')
+        self.files_to_remove.append(fp2)
+        filepath_id = convert_to_id('raw_forward_seqs', 'filepath_type')
+
+        fps = [(fp1, filepath_id), (fp2, filepath_id)]
+
+        filetype_id = get_filetypes()['per_sample_FASTQ']
+        raw_data = RawData.create(filetype_id, [prep_template], fps)
+        params = PreprocessedIlluminaParams(1)
+
+        obs_cmd, obs_output_dir = _get_preprocess_fastq_cmd(raw_data,
+                                                            prep_template,
+                                                            params)
+
+        raw_fps = ','.join([fp for _, fp, _ in
+                            sorted(raw_data.get_filepaths())])
+        exp_cmd = (
+            "split_libraries_fastq.py --store_demultiplexed_fastq -i "
+            "{} --sample_ids 1.SKB8.640193,1.SKD8.640184 -o {} --barcode_type "
+            "golay_12 --max_bad_run_length 3 --max_barcode_errors 1.5 "
```

Comment:
This is because I'm using `PreprocessedIlluminaParams(1)`, I could use the actual value but will need to query the db to get the actual value, thus complicating this test. Note that the test is not actually testing that but that the command is valid ...